core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -249,7 +249,14 @@ private[spark] class ExecutorAllocationManager(
    * yarn-client mode when AM re-registers after a failure.
    */
   def reset(): Unit = synchronized {
-    initializing = true
+    /**
+     * When some tasks need to be scheduled and initial executor = 0, resetting the initializing
+     * field may cause it to not be set to false in yarn.
+     * SPARK-20079: https://issues.apache.org/jira/browse/SPARK-20079
+     */
+    if (maxNumExecutorsNeeded() == 0) {
+      initializing = true
```

Comment:
Sorry but that doesn't really explain much. Why is it bad to ramp up quickly? At which point are things not "initializing" anymore?

Isn't the AM restarting the definition of "I should ramp up quickly because I might be in the middle of a big job being run"?